AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/Mutations.dfy

```diff
@@ -149,6 +166,57 @@ module {:options "/functionSyntax:4" } Mutations {
     return Success(ActiveVerificationHolder.NotDecrypt());
   }
 
+  method VerifyEncryptedHierarchicalKeyForHV2(
+    nameonly item: Types.AwsCryptographyKeyStoreTypes.EncryptedHierarchicalKey,
+    nameonly keyManagerStrategy: KmsUtils.keyManagerStrat,
+    nameonly localOperation: string,
+    nameonly mutationToApply: StateStrucs.MutationToApply
+  )
+    returns (output: Result<ActiveVerificationHolder,Types.Error>)
+
+    requires mutationToApply.ValidState()
+    requires Structure.EncryptedHierarchicalKeyFromStorage?(item)
+    requires KmsArn.ValidKmsArn?(item.KmsArn)
+    requires keyManagerStrategy.ValidState()
+    requires KmsUtils.IsSupportedKeyManagerStrategy(mutationToApply.Terminal.hierarchyVersion, keyManagerStrategy)
+    requires item.Type.ActiveHierarchicalSymmetricVersion? || item.Type.HierarchicalSymmetricVersion?
+    requires mutationToApply.Terminal.hierarchyVersion.v2?
+    modifies keyManagerStrategy.Modifies
+    ensures keyManagerStrategy.ValidState()
+  {
+    var KMSTuple;
+    if localOperation == "ApplyMutation" {
+      KMSTuple := KmsUtils.getEncryptKMSTuple(keyManagerStrategy);
+    } else {
+      KMSTuple := KmsUtils.getDecryptKMSTuple(keyManagerStrategy);
+    }
```

Comment:
Using a string for control is a little fragile... I know I started this trend, but it would be better to create a `type`.

Reply:
Added if..else-if..else with else having failing case.